docs/modules/ROOT/pages/adr/0034-vshnredis-high-availability.adoc

```diff
@@ -0,0 +1,83 @@
+====
+
+== Context
+
+Requirements to the Redis HA architecture:
+
+* Minimizes downtime during failover
+* Supports Kubernetes-native patterns
+* Is operationally manageable
+* Is compatible with our existing infrastructure and client usage patterns
+
+=== Considered Options
+
+**Option 1:  Redis Sentinel + Redis Master/Replica**
+
+image::adr-0034-redis-sentinel.png[Redis Master/Replica & Sentinel Example]
+
+* Description: Traditional Redis HA using Sentinel to monitor and promote replicas when the master fails.
+* Pros:
+  ** Well-established model
+  ** Supported by Redis directly
+  ** Easy to deploy and manage with the Bitnami Helm chart
+  ** Supports Kubernetes-native patterns
+* Cons:
+  ** Requires client that supports sentinel
+  ** Can't be used when exposing Redis outside of Kubernetes (for example Loadbalancer service) and therefore can't be used by Servala.
+
+**Option 2: Redis Cluster Mode**
+
+* Description: Redis Cluster offers built-in sharding and high availability with multiple masters and replicas.
+* Pros:
+  ** Native Redis HA and scalability
+  ** No need for Sentinel
+* Cons:
+  ** Requires client-side cluster awareness
+  ** Higher operational complexity: need to manage multiple masters and replicas
+
+**Option 3: Redis Master/Replica with TCP Proxy in Front (for example HAProxy, Envoy)**
+
+image::adr-0034-redis-sentinel-haproxy.png[Redis Master/Replica & Sentinel & HAProxy Example]
+
+* Description: Deploy Redis in master-replica mode, expose the Redis master behind a TCP load balancer or proxy that can detect failure and route to the promoted master.
+* Pros:
+  ** No need for Sentinel
+  ** Client remains unaware of master failover; proxy handles redirection
+  ** Greater flexibility in failover and health checks
+  ** Can be exposed outside of Kubernetes (for example Loadbalancer service) and therefore be used by Servala
+* Cons:
+  ** Requires custom failover scripts or integration with Sentinel to update routes
+  ** Slight increase in latency due to proxy hop
+  ** Operational complexity: proxy config must track Redis role changes
+
+== Decision
+
+Option 2: Redis Cluster Mode is too complex for our use case at the moment as it requires multiple masters and replicas.
+
+We therefore go with Option 1 and 3. They are simpler alternatives that can be implemented more easily.
+We can even use both options at the same time by using a flag if we want to enable a proxy in front of Redis.
+
+This gives us the flexibility to support users that have sentinel support in their application as well as the users that don't.
+Furthermore, with Option 3 we ensure that the service is also usable with Servala, where the service is exposed publicly.
+
+Last but not least, Options 1 and 3 make it also easy to migrate from HA to non-HA and vice versa. With Option 2 this is not easily doable, as the data is spread across different masters.
```

Comment:
The first part of this section should contain the decision. When I started to read I thought we decided on option 2. Please see https://kb.vshn.ch/app-catalog/adr/0001-architecture-decision-records.html how to structure an ADR document.